public/components/common/seach/autocomplete.tsx

```diff
@@ -0,0 +1,278 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+import './search.scss';
+import React, { createElement, Fragment, useEffect, useRef, useState } from 'react';
+import { render } from 'react-dom';
+import { h } from 'preact';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+import { RAW_QUERY } from '../../../common/constants/explorer';
+import { IQueryBarProps } from './search';
+import { handlePplRequest } from '../../../requests/ppl';
+
+// Possible suggestions (hardcoded)
+const firstCommand = [{ label: 'search' }, { label: 'source' }];
+
+const pipeCommands = [
+  { label: 'dedup' },
+  { label: 'eval' },
+  { label: 'fields' },
+  { label: 'head' },
+  { label: 'rare' },
+  { label: 'rename' },
+  { label: 'sort' },
+  { label: 'stats' },
+  { label: 'top' },
+  { label: 'where' },
+];
+
+const statsCommands = [
+  { label: 'count()' },
+  { label: 'sum()' },
+  { label: 'avg()' },
+  { label: 'max()' },
+  { label: 'min()' },
+];
+
+const indices = [
```

Comment:
How come we are hard-coding the list of indices?

Reply:
I guess it seems like everything is hard-coded. Follow-up question, where will we pull the options from for each of the current hard-coded consts once we have full functionality?

Reply:
Sean is currently working on getting fields and respective values for autocomplete, those works will be added later in the future commits. Previously he was focusing on the UI works, and how to get data for suggestions was not yet determined at that time. Therefore in order to demo the progress, test data was hardcoded for UI functionalities. 

Matched values for a field will be fetched through DSL (Sean is working on). Commands, functions and other PPL related symbols will be fetched through the ppl plugin (opensearch side to-do item).

As for fetching index patterns, it depends on a saved-object solution in order to save/ get all indices for suggestions, which is also a to-do item.